File: 2414/main.go
package main

import "fmt"

func main() {
	fmt.Println(longestContinuousSubstring("abacabc"))	// 3
}

func longestContinuousSubstring(s string) int {
	cur := 1
	ans := 1
	for i := 1; i < len(s); i++ {
		if s[i] == s[i-1] + byte(1) {
			cur++
		} else {
			cur = 1
		}
		ans = max(cur, ans)
	}
	return ans
}

File: 2414/main.rs
struct Solution{}

impl Solution {
    pub fn longest_continuous_substring(s: String) -> i32 {
        let mut cur = 1;
        let mut ans = 1;
        for i in 1..s.len() {
            if s.as_bytes()[i] == s.as_bytes()[i-1] + 1 {
                cur+=1;
            } else {
                cur = 1;
            }
            ans = std::cmp::max(cur, ans);
        }
        return ans
    }
}

fn main() {
    println!("{}", Solution::longest_continuous_substring("abdaabc".to_string()));
}